### ISSUE TITLE
[MPS] embedding_bag(scale_grad_by_freq=True) silently ignores the flag

### ISSUE THREAD

**Issue by bogatyy** [pytorch/pytorch#190061]
### 🐛 Describe the bug

`scale_grad_by_freq=True` must divide each row's weight gradient by the
frequency of its index. The flag reaches
`_embedding_bag_dense_backward_mps` but is never used, so MPS silently
returns the unscaled gradient.

```python
import torch

for dev in ("cpu", "mps"):
    w = torch.randn(4, 3).to(dev).requires_grad_()
    idx = torch.tensor([1, 1], device=dev)
    offs = torch.tensor([0], device=dev)
    out = torch.nn.functional.embedding_bag(idx, w, offs, mode="sum", scale_grad_by_freq=True)
    out.sum().backward()
    print(dev, "row-1 grad:", w.grad[1].tolist())
```

Output on a source build of current main (2.14.0a0+git2662808); also
reproduces on 2.13.0 stable:

```text
cpu row-1 grad: [1.0, 1.0, 1.0]
mps row-1 grad: [2.0, 2.0, 2.0]
```

The flag arrives at
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/EmbeddingBag.mm#L185-L194
and is unused in the params or kernel dispatch. #185119 fixed the identical
silently-ignored flag for `nn.Embedding` on MPS; `nn.EmbeddingBag` was left
out.

### Versions

PyTorch 2.14.0a0+git2662808 (source build of main) and 2.13.0; macOS 26.5.2 (arm64), Apple M5 Max, Python 3.12.13.

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93 @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by CodersAcademy006** [comment]
Root cause is that `_embedding_bag_dense_backward_mps` (`aten/src/ATen/native/mps/operations/EmbeddingBag.mm`, ~L185) accepts `scale_grad_by_freq` as a parameter (~L191) but the body never reads it — it sets up strides and per-sample-weight params and dispatches, with no frequency-count computation at all. The CPU backward builds a `counts` array and divides each row's gradient by `counts[index]`; the MPS backward has no equivalent, so the flag is silently a no-op. Fixing it means computing per-index frequencies (as CPU does) and scaling the accumulated weight gradient before the kernel returns. This pairs with #190063 (the CPU path scales by the wrong count) — together the `scale_grad_by_freq` handling is wrong on CPU and absent on MPS. @bogatyy
